### PR TITLE
Create types for pg-database-url

### PR DESCRIPTION
**Pull request by tdukart**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/pg-database-url/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    database: string;
+    host?: string;
+    password?: string;
+    port?: number;
+    username: string;
+}
+
+declare function pg_database_url(config: PgDatabaseUrlConfig): string;
+
+export = pg_database_url;
```
Comment (martin-badin) [inline types/pg-database-url/index.d.ts:16]:
```suggestion
declare namespace pgDatabaseUrl {
    interface Config {
        database: string;
        host?: string;
        password?: string;
        port?: number;
        username: string;
    }
}

declare function pgDatabaseUrl(config: pgDatabaseUrl.Config): string;

export = pgDatabaseUrl;
```

**Review by martin-badin — CHANGES_REQUESTED** [review]

**Review by martin-badin — APPROVED** [review]
Looks good

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@tdukart thx!

**Comment by tdukart** [comment]
Ready to merge

**Pull request merged**